geomesa-kafka/geomesa-kafka-datastore/src/main/scala/org/locationtech/geomesa/kafka/KafkaDataStoreSchemaManager.scala

```diff
@@ -61,10 +62,37 @@ trait KafkaDataStoreSchemaManager extends DataStore  {
       AdminUtils.createTopic(zkClient, kfc.topic, partitions, replication)
     }
 
+    // Ensure the topic is ready before adding the new layer to the SFT cache
+    waitUntilTopicReady(zkClient, kfc.topic, partitions)
+
     // put it in the cache
     schemaCache.put(typeName, kfc)
   }
 
+  // Wait until a topic is ready for writes.
+  // NB: The assumption here is that waiting for the leader to be elected is sufficient.
+  // Further, this function delegates to a function which waits for a leader for each partition.
+  def waitUntilTopicReady(zkClient: ZkClient, topic: String, partitions: Int, timeToWait: Long = 5000L): Unit = {
```

Comment:
I don't mind it being per partition.  I'm assuming that the partitions will be getting sorted out together, and generally this should get handled within 5 seconds.

There are several assumptions and I'm happy for push back.  If this is sufficient for a first pass, then I'm happy.